hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerSizeCountTask.java

```diff
@@ -122,6 +123,11 @@ private void process(ContainerInfo container,
       Map<ContainerSizeCountKey, Long> map) {
     final ContainerID id = container.containerID();
     final long currentSize = container.getUsedBytes();
+    if (currentSize < 0) {
+      LOG.error("Negative container size: {} for container: {}", currentSize,
+          id);
+      return;
+    }
```

Comment:
This will just print the error in a different message instead of fixing it.  Why the `currentSize` could be negative?

Reply:
While trying to understand the reason for this underlying abnormality. What I have noticed from the recon logs of the cluster that reported the issue is that the containerSizeCountTask detected Negative size for Container IDs: `[ID: 3979, Size: -63] [ID: 3985, Size: -63] [ID: 13509, Size: -33554432]`

Doing a container info on these container ID’s I noticed that these containers have been in the DELETED state. If the container size is being fetched after the container is marked for deletion, could there be a scenario where the size is momentarily calculated or reported as negative??
 possibly due to how the deletion process affects the metadata

Reply:
> ... could there be a scenario where the size is momentarily calculated or reported as negative??

It probably is a bug.  Let's don't change the log message since it will hide the bug instead of fixing it.

Reply:
I mean we should keep the `IllegalArgumentException` instead adding this `LOG.error` message.

The correct way is to fix the datanodes (i.e. to not to report negative used bytes).  Then, the `IllegalArgumentException` will disappear.

Reply:
@ArafatKhan2198 , we may treat -ve size as zero in `ContainerSizeCountTask`.
```java
    final long usedBytes = container.getUsedBytes();
    final long currentSize;
    if (usedBytes < 0) {
      LOG.warn("Negative usedBytes ({}) for container {}, treating it as 0", usedBytes, id);
      currentSize = 0;
    } else {
      currentSize = usedBytes;
    }
```